setup.py

```diff
@@ -41,6 +49,57 @@ def find_version(*file_paths):
     raise RuntimeError("Unable to find version string.")
 
 
+cmdclass = {}
+if SPHINX_AVAILABLE:
+
+    class MakeDocs(BuildDoc):
+        """Port of old `cylc-make-docs`. Removed in #2989.
+
+        This class extends the Sphinx command class for setuptools. With
+        a difference that it tries to mimic the behaviour of the previous
+        `cylc-make-docs`.
+
+        So first it will execute `make-commands.sh`, which builds the
+        commands help information, in the appendices.
+
+        Then, instead of calling one builder, this class will call the
+        builder fot he single HTML, and also the builder for the multiple
+        HTML documentation.
+
+        Finally, one more tweak in this class is to move the doctrees
+        folder (in the same level as the documentation) to within the
+        documentation folder, named `.doctrees`, as before with
+        `cylc-make-docs`.
+        """
+
+        def run(self):  # type: () -> None
+            try:
+                self.spawn(["./doc/src/custom/make-commands.sh"])
+            except DistutilsExecError as e:
+                self.warn("Failed to run make-commands.sh")
+                raise e
+            self.do_run("html", "built-sphinx")
+            self.do_run("singlehtml", "built-sphinx-single")
+
+        def do_run(self, builder: str, output_dir: str):
+            """
+            Args:
+                builder (str): name of the Sphinx builder
+                output_dir (str): directory to write the documentation produced
+            """
+            self.builder = builder
+            self.builder_target_dirs = [
+                (builder, join(self.build_dir, output_dir))]
+            super().run()
+            # move doctrees to $build_dir/.doctrees
+            correct_doctrees = join(self.builder_target_dirs[0][1],
+                                    ".doctrees")
+            rmtree(correct_doctrees, ignore_errors=True)
+            move(self.doctree_dir, correct_doctrees)
+
+    cmdclass["build_sphinx"] = MakeDocs
```

Comment:
We will need to think about how we are going to implement remaining docs-building functionality into this framework e.g. building docs for multiple versions of Cylc.

Reply:
Yup, agreed. And had a bit of struggle with Sphinx + setuptools. Even though we can call multiple builders with sphinx command line, its setuptools integration is limited, and allows only a single one.

It seems like the setuptools part is not well maintained, and seems to be using some old code style from sphinx.